packages/devtools_app/lib/src/tree.dart

```diff
@@ -177,6 +214,9 @@ mixin TreeMixin<T extends TreeNode<T>> {
     for (T root in roots) {
       traverse(root, (n) {
         if (onTraverse != null) onTraverse(n);
+        if (!n.shouldShow()) {
+          return false;
+        }
```

Comment:
can you add a test for a case that would have been incorrect without this check?

Reply:
Not 100% sure why this was necessary. Doesn't seem to break anything if I remove it, so I'll just do that.